src/Product/ProductHelper.php

```diff
@@ -332,6 +334,81 @@ public function is_product_synced( WC_Product $product ): bool {
 		return ! empty( $synced_at ) && ! empty( $google_ids );
 	}
 
+	/**
+	 * Indicates if a product is ready for sending Notifications.
+	 * A product is ready to send notifications if DONT_SYNC_AND_SHOW is not enabled and the post status is publish.
+	 *
+	 * @param WC_Product $product
+	 *
+	 * @return bool
+	 */
+	public function is_ready_to_notify( WC_Product $product ): bool {
+		return ChannelVisibility::DONT_SYNC_AND_SHOW !== $this->get_channel_visibility( $product ) && $product->get_status() === 'publish';
+	}
+
+	/**
+	 * Indicates if a product is ready for sending a create Notification.
+	 * A product is ready to send create notifications if is ready to notify and has not sent create notification yet.
+	 *
+	 * @param WC_Product $product
+	 *
+	 * @return bool
+	 */
+	public function should_trigger_create_notification( WC_Product $product ): bool {
+		return $this->is_ready_to_notify( $product ) && ! $this->has_notified_creation( $product );
+	}
+
+	/**
+	 * Indicates if a product is ready for sending an update Notification.
+	 * A product is ready to send update notifications if is ready to notify and has sent create notification already.
+	 *
+	 * @param WC_Product $product
+	 *
+	 * @return bool
+	 */
+	public function should_trigger_update_notification( WC_Product $product ): bool {
+		return $this->is_ready_to_notify( $product ) && $this->has_notified_creation( $product );
+	}
+
+	/**
+	 * Indicates if a product is ready for sending a delete Notification.
+	 * A product is ready to send delete notifications if it is not ready to notify and has sent create notification already.
+	 *
+	 * @param WC_Product $product
+	 *
+	 * @return bool
+	 */
+	public function should_trigger_delete_notification( WC_Product $product ): bool {
+		return ! $this->is_ready_to_notify( $product ) && $this->has_notified_creation( $product );
+	}
+
+	/**
+	 * @param WC_Product $product
+	 *
+	 * @return bool
+	 */
+	public function has_notified_creation( WC_Product $product ): bool {
```

Comment:
> I got a bit confused with all the statuses but what I understand is that these statuses indicate whether the product has been created/synced, helping us, for example, to avoid sending an update/delete topic if the product has not been notified yet.

Correct. For example. Imagine this scenario:

- Creating a product with DONT SYNC AND SHOW won't trigger `product.create` notification. 
- Then the user updates the product and set SYNC AND SHOW. The notification should be `product.create` not `product.update` for doing this is necessary the metas and some helper functions.

Reply:
> Another scenario involves an existing product that is already synced with the MC; the created metadata does not exist because the product was already synced before we started using notifications. If the merchant updates the product, we will send the create notification, which is incorrect because the product was already created in the MC. A similar case would happen when removing a product

I assume we don't take in consideration the previous statuses of (PUSH based) MC for sending the Notifications

Reply:
I will move that question to the P2